README.md

```diff
@@ -1,2 +1,20 @@
-# gateway
-Gateway for sending firebase events
+# Project Title
+
+Gateway for Firebase notifications
+
+
+## Getting Started
+
+
+### Installing
+Create a Python 3.6 virtual environment and install the packages:
+    
+    pip install virtualenv
+    virtualenv venv
+    source venv/bin/activate
+    pip install -r requirements.txt
+
+```
+Run
+./app.py
+```
```

Comment:
About 

> * Function description yes/no? If Yes what is the best way of doing a function description.

I think the README is a good place to start. After installing and spinning it up; how does it work actually?! Then when it becomes too big and complex to present in a README, move it to a separate document. And when it's as big and complex as Passport or Analytics4, give it its own Sphinx site like http://docs.gunicorn.org 🍔

Reply:
I was more thinking of the functions/prodcedures (def ....) shall be documented. 
I guess type hinting replaces the need for that to some extent.